src/pointcloud_to_laserscan_nodelet.cpp

```diff
@@ -198,9 +200,17 @@ void PointCloudToLaserScanNodelet::cloudCb(const sensor_msgs::PointCloud2ConstPt
     cloud_out = cloud_msg;
   }
 
+  
+  if (keep_original_angle_frame_) {
+    cloud_iterator = cloud_msg;
```

Comment:
When the condition is true, we are iterating over the original cloud instead of the transformed one. This will work nicely for the angle check, but the range check will also be done from the original frame.